src/main/java/gregtech/common/metatileentities/multi/MetaTileEntityFissionReactor.java

```diff
@@ -0,0 +1,1104 @@
+        }
+
+        return !(potentialTile instanceof IFissionReactorHatch || potentialTile instanceof IMaintenanceHatch);
+    }
+
+    protected EnumFacing getUp() {
+        return RelativeDirection.UP.getRelativeFacing(frontFacing, upwardsFacing, isFlipped);
+    }
+
+    protected EnumFacing getRight() {
+        return RelativeDirection.RIGHT.getRelativeFacing(frontFacing, upwardsFacing, isFlipped);
+    }
+
+    /**
+     * Uses the upper layer to determine the diameter of the structure
+     */
+    protected int findDiameter(int heightAbove) {
+        int i = 1;
+        while (i <= 15) {
+            if (this.isBlockEdge(this.getWorld(), this.getPos().offset(getUp(), heightAbove),
+                    this.getFrontFacing().getOpposite(),
+                    i))
+                break;
+            i++;
+        }
+        return i;
+    }
+
+    /**
+     * Checks for casings on top or bottom of the controller to determine the height of the reactor
+     */
+    protected int findHeight(boolean top) {
+        int i = 1;
+        while (i <= 15) {
+            if (this.isBlockEdge(this.getWorld(), this.getPos(), top ? getUp() : getUp().getOpposite(), i))
+                break;
+            i++;
+        }
+        return i - 1;
+    }
+
+    @Override
+    public void updateFormedValid() {
+        // Take in coolant, take in fuel, update reactor, output steam
+
+        if (!this.getWorld().isRemote && this.getOffsetTimer() % 20 == 0) {
+            if (this.lockingState == LockingState.LOCKED) {
+                // Coolant handling
+                if (this.getOffsetTimer() % 100 == 0) {
+                    if (isFlowingCorrectly) {
+                        if (getWorld().rand.nextDouble() > (1 - 0.01 * this.getNumMaintenanceProblems())) {
+                            isFlowingCorrectly = false;
+                        }
+                    } else {
+                        if (getWorld().rand.nextDouble() > 0.12 * this.getNumMaintenanceProblems()) {
+                            isFlowingCorrectly = true;
+                        }
+                    }
+                }
+
+                // Fuel handling
+                if (this.fissionReactor.isDepleted()) {
+                    boolean canWork = true;
+                    boolean needsReset = false;
+                    for (IFuelRodHandler fuelImport : this.getAbilities(MultiblockAbility.IMPORT_FUEL_ROD)) {
+                        if (fuelImport.getStackHandler().extractItem(0, 1, true).isEmpty()) {
+                            canWork = false;
+                            this.lockingState = LockingState.MISSING_FUEL;
+                            break;
+                        } else if (!((MetaTileEntityFuelRodImportBus) fuelImport).getExportHatch(this.height - 1)
+                                .getExportItems().insertItem(0,
+                                        OreDictUnifier.get(OrePrefix.fuelRodHotDepleted, fuelImport.getFuel()), true)
```

Comment:
This should use its own registry class which uses maps with the appropriate `ItemStackHashStrategy` to store items. It should be used to associate inputs and outputs instead of using ore prefixes.

This registry should have api methods to associate different arbitrary itemstacks as all the relevant fuel items, and everything should go off of that instead. This current system is brittle and hard to extend.

The registry also needs to display these associations in JEI. Right now there is no way to go from regular -> depleted or the reverse in JEI which makes navigating these recipes hard.